core/templates/dev/head/editor/SidebarStateParameterChanges.js

```diff
@@ -20,30 +20,192 @@
  */
 
 oppia.controller('StateParamChangesEditor', [
-    '$scope', 'editorContextService', 'changeListService', 'explorationStatesService',
-    function($scope, editorContextService, changeListService, explorationStatesService) {
+    '$scope', 'editorContextService', 'stateParamChangesService',
+    'editabilityService', 'explorationParamSpecsService', 'warningsData',
+    function(
+      $scope, editorContextService, stateParamChangesService,
+      editabilityService, explorationParamSpecsService, warningsData) {
 
-  $scope.$on('refreshStateEditor', function() {
-    $scope.initStateParamChangesEditor();
+  $scope.stateParamChangesService = stateParamChangesService;
+  $scope.editabilityService = editabilityService;
+  $scope.isStateParamChangesEditorOpen = false;
+
+  $scope.warningText = '';
+
+  var _INVALID_PARAMETER_NAMES = GLOBALS.INVALID_PARAMETER_NAMES;
+
+  $scope.$on('stateEditorInitialized', function(evt, stateData) {
+    $scope.stateParamChangesMemento = null;
+
+    var stateName = editorContextService.getActiveStateName();
+    stateParamChangesService.init(
+      stateName, stateData.param_changes, stateData, 'param_changes');
   });
 
-  $scope.initStateParamChangesEditor = function() {
-    $scope.stateName = editorContextService.getActiveStateName();
-    var stateData = explorationStatesService.getState($scope.stateName);
-    $scope.stateParamChanges = stateData.param_changes || [];
+  $scope.PREAMBLE_TEXT = {
+    'Copier': 'to',
+    'RandomSelector': 'to one of'
+  };
+
+  var _DEFAULT_PARAM_SPEC = {
+    obj_type: 'UnicodeString'
+  };
+
+  // The name of this parameter change needs to be filled in.
+  var _DEFAULT_PARAM_CHANGE = {
+    name: '',
+    customization_args: {
+      value: '5',
```

Comment:
We only have one type of parameter at the moment (UnicodeString), so I decided to simplify.

(Edit: Discussed offline; decided to leave as-is.)